dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +75,83 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__catchInsecurePingResponse: function(httpsArgs) {
+		var httpsXhr = httpsArgs[1], self = this,
+			handleSuccess, handleError;
+
+		handleSuccess = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			// https did not work but http did...something is wrong with the cert
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+		};
+		handleError = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			if (httpXhr.status === 0) {
+				// https is failing as well...service is unavailable
+				self.Dispatcher.handleAppEvent({
+					name: "SERVICE_UNAVAILABLE",
+					status: httpXhr.status
+				});
```

Comment:
Hmm, we'll also want to return a reject here so the service unavailable view doesn't redirect to login